x/wasm/internal/keeper/query_plugins.go

```diff
@@ -146,6 +164,71 @@ func NoCustomQuerier(sdk.Context, json.RawMessage) ([]byte, error) {
 	return nil, wasmvmtypes.UnsupportedRequest{Kind: "custom"}
 }
 
+func IBCQuerier(wasm *Keeper, channelKeeper types.ChannelKeeper) func(ctx sdk.Context, caller sdk.AccAddress, request *wasmvmtypes.IBCQuery) ([]byte, error) {
+	return func(ctx sdk.Context, caller sdk.AccAddress, request *wasmvmtypes.IBCQuery) ([]byte, error) {
+		if request.PortID != nil {
+			contractInfo := wasm.GetContractInfo(ctx, caller)
+			res := wasmvmtypes.PortIDResponse{
+				PortID: contractInfo.IBCPortID,
+			}
+			return json.Marshal(res)
+		}
+		// I just discovered we cannot query by port (how odd since they store it like this)
+		// Anyway, I will allow Port as an optional filter
+		// TODO: let's revisit this query (if needed, if we should change it)
```

Comment:
The code looks good. Is gas cost your concern with the iterator?
I wonder if we can abort the iterator early when > portID ?

Let's avoid todos in code and have an issue instead.

Reply:
Gas cost is my issue. This could become expensive and is always O(total channels in the system). 

My main note was that our query does not align with what the sdk exposed, so I'd rather redefine this to never take a port (since we cannot filter by port) and then it is just as inefficient as the description is. But I wanted feedback before changing behavior.

Reply:
If we can not abort the iterator early then I agree that there is no benefit to filter. Let's go for the simplest solution.